### PR TITLE
Reload the page after webpack watch compile

### PR DESCRIPTION
**Pull request by atimmer**
I tried implementing webpack-dev-server but that I didn't crack that this time. This PR will at least save everyone some F5/Ctrl+R keystrokes.

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -177,6 +177,13 @@ function gutenberg_register_scripts_and_styles() {
 		filemtime( gutenberg_dir_path() . 'blocks/build/edit-blocks.css' )
 	);
 	wp_style_add_data( 'wp-edit-blocks', 'rtl', 'replace' );
+
+	if ( WP_DEBUG ) {
```
Comment (aduth) [inline lib/client-assets.php]: Would `SCRIPT_DEBUG` be the more appropriate constant here? Based on the description, it seems primarily aimed at toggling minified version, so perhaps not.

https://codex.wordpress.org/Debugging_in_WordPress#SCRIPT_DEBUG

**Review by aduth — APPROVED** [review]
I'd occasionally find it useful to keep an outdated version of an editor tab active for comparison's sake, but not often enough that this would disrupt my typical workflow.

I do worry a bit since Webpack rebuilds can be very noisy that we could encounter annoyances around unexpected reloads.

Generally I'd be okay with something like this though.

>I tried implementing webpack-dev-server but that I didn't crack that this time.

Can you elaborate on the issues you were encountering?

**Comment by atimmer** [comment]
@aduth Because WordPress itself is served by a different web server you have to proxy every request. Once I had that in place I didn't get any further. Webpack dev server wouldn't show console messages that it is trying to reload and the page wouldn't reload at all.

If you want we could implement a new Gutenberg specific constant like `GUTENBERG_LIVE_RELOAD` which could be used to disable live reload.

**Comment by atimmer** [comment]
This is ready for review.

Live reloading can now be enabled by setting the `GUTENBERG_LIVE_RELOAD` constant to `true`. Any value other than `true` will be used as the URL to load. This can be used to customize the URL.

I've also made it possible to customize the port that the livereloading server uses. This is useful when multiple projects support livereloading. This is used like so: `GUTENBERG_LIVE_RELOAD_PORT=35730 yarn dev`.

**Comment by gziolo** [comment]
@atimmer - it was accepted and seems to be useful, can you update and merge? I like that it is optional 👍

**Review by ntwb — APPROVED** [review]

**Comment by gziolo** [comment]
This could also be helpful to include some note how to use it in [CONTRIBUTING document](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md).

**Comment by pento** [comment]
I'm cool with this going in, but I'd want to actually build some stuff with it enabled before we start recommending it to all contributors.

**Pull request merged**